javascripts/2048.js

```diff
@@ -1,32 +1,406 @@
+    var typeValue = tile.getAttribute(type);
+    // this gets the value of the opposite type (col for up/down, row for R/L)
+    var oppositeType = (type == "data-row") ? "data-col" : "data-row";
+    var oppositeValue = tile.getAttribute(oppositeType);
+    // find neighbor value ( if c1,r1 and moving up, neighbor is c1, r2)
+    var neighborValue = parseInt(typeValue) - magnitude;
+    // nasty block text stuff
+    var neighborText = ".tile[" + type + "=\"" + neighborValue + "\"][" + oppositeType + "=\"" + oppositeValue + "\"]";
+    // use block text to check if neighbor exists
+    var neighbor = $(neighborText);
+
+    if (neighbor.length > 0) {
+      neighbor = neighbor[0];
+      if (neighbor.getAttribute("data-val") == dataVal) {
+        return neighbor;
+      }
+    } else {
+      return null;
+    }
+  }
+
+  function mergeTiles() {
+    var sortedTiles = orderTiles();
+
+    for (var i = 0; i < sortedTiles.length; i++) {
+      var neighbor = findMergeableTile(sortedTiles[i], turnType, turnMagnitude);
+      // if neighbor exists, then double current tile's value and delete neighbor tile
+      if (neighbor) {
+        var currentVal = parseInt(sortedTiles[i].getAttribute("data-val"));
+        var newVal = currentVal * 2;
+
+        // setting neighbor tile's to the original tile's position for animation
+        neighbor.className = neighbor.className + " merged";
+        neighbor.setAttribute("data-row", sortedTiles[i].getAttribute("data-row"));
+        neighbor.setAttribute("data-col", sortedTiles[i].getAttribute("data-col"));
+
+        sortedTiles[i].setAttribute("data-val", (newVal));
+        updateScore(newVal);
+        sortedTiles[i].innerHTML = (newVal);
+        // add animation to sortedtile[i] here
+        sortedTiles[i].className = sortedTiles[i].className + " blowup";
+        var mergedTile = $(".tile.blowup");
+        mergedTile.on("animationend", function(){
+            mergedTile.removeClass("blowup");
+        });
+
+        var neighborIndex = sortedTiles.indexOf(neighbor);
+        // so sorry
+        sortedTiles = sortedTiles.splice(0,neighborIndex).concat(
+           sortedTiles.splice(1, sortedTiles.length-1)
+           );
+        checkWin(newVal);
+
+      }
+
+      // try adding document.addEventListener("transitionend", function(){})
+      // look for delegation syntax
+      // --> in jquery, to make the transition end more deliberate
+      // animation to hide the merged tile, then delete it
+      $(".tile.merged").remove();
+      // $(".tile.merged").on("transitionend", function() {
+        // $(this).hide("fast", function(){
+      //   $(this).remove();
+      //   // });
+      // });
+    }
+  }
+
+  function rowOrColumn(direction) {
+    var type = "";
+    if (direction == 38 || direction == 40) {
+      type = "data-row";
+    } else if (direction == 37 || direction == 39) {
+      type = "data-col";
+    }
+
+    return type;
+  }
+
+  function getMagnitude(direction) {
```

Comment:
It seems that all calls to `getMagnitude` are followed by `parseInt`. Maybe `getMagnitude` could directly return an integer instead of a string?